### PR TITLE
refactor: merge sovran into core lib

### PR DESCRIPTION
**Pull request by oscb**
BREAKING CHANGE: @segment/sovran-react-native is now deprecated. Remove from your dependencies. All functionality is now rolled in the core package.

Required changes:

- Remove `@segment/sovran-react-native` from your dependencies.

- iOS: If you are using `trackDeeplinks` or `setAnonymousId` from native you'll need to update the imports to

```diff
- #import <segment_analytics_react_native-Swift.h>
+ #import "SegmentAnalyticsReactNative-Swift.h"
```

### CODE REVIEW

File: examples/E2E-73/android/app/src/main/java/com/analyticsreactnativee2e73/MainApplication.kt [hunk 1]
```diff
@@ -11,14 +12,14 @@ import com.facebook.react.defaults.DefaultReactHost.getDefaultReactHost
...
 import com.facebook.soloader.SoLoader
+import com.segmentanalyticsreactnative.AnalyticsReactNativePackage
 
 class MainApplication : Application(), ReactApplication {
 
   override val reactNativeHost: ReactNativeHost =
       object : DefaultReactNativeHost(this) {
         override fun getPackages(): List<ReactPackage> {
           // Packages that cannot be autolinked yet can be added manually here, for example:
-          // packages.add(new MyReactNativePackage());
```
Comment (chrisradek) [inline examples/E2E-73/android/app/src/main/java/com/analyticsreactnativee2e73/MainApplication.kt:21]: Did you intentionally remove this? Above line reads oddly now.

Comment (oscb) [reply examples/E2E-73/android/app/src/main/java/com/analyticsreactnativee2e73/MainApplication.kt:21]: Should reset this thing. This is from the boilerplate of RN example apps.

File: examples/README.md [hunk 2]
```diff
@@ -25,31 +25,30 @@ That said you still have to setup some files to make it work.
 
 First create a file (e.g. `workspace.js`, but naming doesn't matter) to initialize the helper. The `getLinkedData` function receives an array of relative paths (from the `packages` directory) to link. e.g. `core`, `sovran` or for specific plugins `plugins/plugin-idfa`.
```
Comment (chrisradek) [inline examples/README.md:26]:
```suggestion
First create a file (e.g. `workspace.js`, but naming doesn't matter) to initialize the helper. The `getLinkedData` function receives an array of relative paths (from the `packages` directory) to link. e.g. `core` or for specific plugins `plugins/plugin-idfa`.
```

File: packages/core/tsconfig.json [hunk 3]
```diff
@@ -5,6 +5,5 @@
     "rootDir": ".",
     "baseUrl": "."
   },
-  "references": [{ "path": "../sovran" }],
```
Comment (chrisradek) [inline packages/core/tsconfig.json:8]: Looks like `tsconfig.linter.json` also has this reference

**Review by chrisradek — COMMENTED** [review]
Overall looks good, had a few minor comments.

Also wondering, do we still need to mock Sovran here:
https://github.com/segmentio/analytics-react-native/blob/7160d3d669502b390e9d6712c7db39b30c410e91/packages/shared/__mocks__/react-native.ts#L36-L40

File: packages/core/tsconfig.json (hunk 3 above)
Comment (oscb) [reply packages/core/tsconfig.json:8]: Good catch!

**Comment by oscb** [comment]
> Overall looks good, had a few minor comments.
> 
> Also wondering, do we still need to mock Sovran here:
> 
> https://github.com/segmentio/analytics-react-native/blob/7160d3d669502b390e9d6712c7db39b30c410e91/packages/shared/__mocks__/react-native.ts#L36-L40

No need for that anymore. It's part of the methods that I removed altogether

**Review by chrisradek — APPROVED** [review]

**Pull request merged**